hedera-node/docs/design/services/token-service/airdrops/token-claim-airdrop.md

```diff
@@ -0,0 +1,165 @@
+# Introduce TokenClaimAirdrop transaction
+## Purpose
+
+We need to add a new functionality that would make it possible for an airdrop receiver to accept a pending airdrop transfer. This would be the only way for a receiver, which hasn't been associated to a given token to accept an airdropped token that has been in pending airdrops state.
+
+## Goals
+
+1. Define new `TokenClaimAirdrop` HAPI transaction
+2. Implement token claim airdrop transaction handler logic
+
+## Non-Goals
+
+- Implement token claim airdrop in system contract functions
+
+## Architecture
+
+The implementation related to the new `TokenClaimAirdrop` transaction will be gated behind a feature flag.
+
+### HAPI updates
+
+Create new transaction type as defined in the HIP:
+
+```protobuf
+/**
+ * Token claim airdrop<br/>
+ * Complete one or more pending transfers on behalf of the recipient(s) for each airdrop.<br/> 
+ * The sender MUST have sufficient balance to fulfill the airdrop at the time of claim. If the
+ * sender does not have sufficient balance, the claim SHALL fail.
+ * 
+ * Each pending airdrop successfully claimed SHALL be removed from state and SHALL NOT be available
+ * to claim again.
+ * 
+ * Each claim SHALL be represented in the transaction body and SHALL NOT be restated
+ * in the record file.<br/>
+ * All claims MUST succeed for this transaction to succeed.
+ */
+message TokenClaimAirdropTransactionBody {
+    /**
+     * A list of one or more pending airdrop identifiers.<br/>
+     * This transaction MUST be signed by the account referenced in the `receiver_id` for
+     * each entry in this list.
+     * <p>
+     * This list MUST contain between 1 and 10 entries, inclusive.<br/>
+     * This list MUST NOT have any duplicate entries.
+     */
+    repeated PendingAirdropId pending_airdrops = 1;
+}
+
+/**
+ * A unique, composite, identifier for a pending airdrop.
+ *
+ * Each pending airdrop SHALL be uniquely identified by a PendingAirdropId.
+ * A PendingAirdropId SHALL be recorded when created and MUST be provided in any transaction
+ * that would modify that pending airdrop (such as a `claimAirdrop` or `cancelAirdrop`).
+ */
+message PendingAirdropId {
+    /**
+     * A sending account.<br/>
+     * This is the account that initiated, and SHALL fund, this pending airdrop.<br/>
+     * This field is REQUIRED.
+     */
+    AccountID sender_id = 1;
```

Comment:
No, for claim the payer can be different account from the sender or the receiver.
There is an example in the HIP that mentions this:
```
An account with no open auto-association slots can receive airdrops but must send a TokenClaimAirdropTransaction, which requires a payer.
If the account has zero hbars, then it can still claim the transfer if someone else is willing to pay for that transaction.
For example, a Dapp could be the payer on the transaction.
Both the Dapp and the account must sign the transaction.
```